ImageNet/train/google_resnet50_tpu_tensorflow17.json

```diff
@@ -0,0 +1,27 @@
+{
+  "version": "v1.0",
+  "author": "Google Cloud TPU",
+  "authorEmail": "cloud-tpu-dawnbench@googlegroups.com",
+  "framework": "TensorFlow v1.7rc1",
+  "codeURL": "https://github.com/tensorflow/tpu/tree/b189540102d6b5b40b1730d7e5ad5c884bae323c/models/experimental/resnet_bfloat16",
+  "model": "ResNet50",
+  "hardware": "GCP n1-standard-2, Cloud TPU",
+  "costPerHour": 6.595,
+  "timestamp": "2018-03-25",
+  "misc": {
+    "optimizer": "SGD with Momentum",
+    "momentum": 0.9,
+    "weightDecay": 0.0001,
+    "batchSize": 1024,
+    "schedule": [
+      {"learning_rate": 0.08, "epochs": 1},
+      {"learning_rate": 0.16, "epochs": 1},
+      {"learning_rate": 0.24, "epochs": 1},
+      {"learning_rate": 0.32, "epochs": 1},
+      {"learning_rate": 0.4, "epochs": 25},
```

Comment:
This is minor, but your TSV has a total of 95 epochs and this only adds up to 94. I think the schedule should either be: 

```
    "schedule": [
      {"learning_rate": 0.067, "epochs": 1},
      {"learning_rate": 0.133, "epochs": 1},
      {"learning_rate": 0.2, "epochs": 1},
      {"learning_rate": 0.267, "epochs": 1},
      {"learning_rate": 0.333, "epochs": 1},
      {"learning_rate": 0.4, "epochs": 25},
      {"learning_rate": 0.04, "epochs": 30},
      {"learning_rate": 0.004, "epochs": 20},
      {"learning_rate": 0.0004, "epochs": 15}
    ]
```

if `current_epoch` starts at 0 Or 

```
    "schedule": [
      {"learning_rate": 0.08, "epochs": 1},
      {"learning_rate": 0.16, "epochs": 1},
      {"learning_rate": 0.24, "epochs": 1},
      {"learning_rate": 0.32, "epochs": 1},
      {"learning_rate": 0.4, "epochs": 26},
      {"learning_rate": 0.04, "epochs": 30},
      {"learning_rate": 0.004, "epochs": 20},
      {"learning_rate": 0.0004, "epochs": 15}
    ]
```

if `current_epoch` starts at 1

From the code, it looks like the former is more correct because [`current_epoch` is just `global_step` divided by `batches_per_epoch`](https://github.com/tensorflow/tpu/blob/b189540102d6b5b40b1730d7e5ad5c884bae323c/models/experimental/resnet_bfloat16/resnet_main.py#L300).

Reply:
I think the last entry with learning_rate 0.0004 needs to be 16 instead of 15 as that makes the learning rate correct.

Reply:
Actually it just seems  to be running at learning rate 0 for the first epoch which doesn't update the weights but batch norm is still in place and provides the linear learning rate scaling as well.

Reply:
Sorry for the confusion, the 0 tripped me for a bit.

Reply:
I don't think there is a perfect way to map your warm-up procedure to the `schedule` because your code updates the learning per step and not per epoch. I think the closes equivalent would be the first option I had:

```
    "schedule": [
      {"learning_rate": 0.067, "epochs": 1},
      {"learning_rate": 0.133, "epochs": 1},
      {"learning_rate": 0.2, "epochs": 1},
      {"learning_rate": 0.267, "epochs": 1},
      {"learning_rate": 0.333, "epochs": 1},
      {"learning_rate": 0.4, "epochs": 25},
      {"learning_rate": 0.04, "epochs": 30},
      {"learning_rate": 0.004, "epochs": 20},
      {"learning_rate": 0.0004, "epochs": 15}
    ]
```

Running for a full epoch with a learning rate of 0.0 would be a waste, so I think it is worth updating to the schedule above. 

Also, to make this clearer and easier to reproduce, please add another a `warmUp` field to `misc` with a textual description. Maybe something like:

```
"misc": {
  "warmUp": "the learning rate starts at 0, then it increases linearly per iteration for the first 5 epochs",
  ...
}
```

This is basically the comment on [L167](https://github.com/tensorflow/tpu/blob/b189540102d6b5b40b1730d7e5ad5c884bae323c/models/experimental/resnet_bfloat16/resnet_main.py#L167), but it is useful to draw attention to if someone is having trouble with the per epoch schedule.

Reply:
Added the warmup section in misc. I can look into the updated schedule and submit a follow up PR if needed.

Reply:
Thanks. That sounds fair.